src/Lecture2.hs

```diff
@@ -85,7 +98,12 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+
+evenLists :: [[a]] -> [[a]]
+evenLists = filter isOfEvenLength
+
+isOfEvenLength :: [a] -> Bool
+isOfEvenLength list = if mod (length list) 2 == 0 then True else False
```

Comment:
There's a standard function `even` in Haskell that checks if the number is even. Using this function and function composition, you can implement `evenLists` in a short an elegant way 👌🏻